### PR TITLE
Add a check for existing config

### LINKED ISSUE #35: [Feature Request] Check if config already exists before creating it
Currently, on init the config file is always written even if it already exists, potentially overriding saved user config.

Please add a file exists check before: https://github.com/LintaoAmons/scratch.nvim/blob/9b35cf9ff22f243cea72febc37416897d6eae184/lua/scratch/config.lua#L166

### PR DESCRIPTION
**Pull request by Rizhiy**
Fixes #35

### CODE REVIEW

**Comment by LintaoAmons** [comment]
Hi, I'm not very sure if we need add this functionality, cause currently these's a command `ScratchInitConfig`, will call this function

https://github.com/LintaoAmons/scratch.nvim/blob/9b35cf9ff22f243cea72febc37416897d6eae184/lua/scratch/config.lua#L250

And it should be able to reset the configuration.

But yes, it's quit dangerous to rewrite the configuration without warning the user. Previously I was thinking maybe the configuration is tracked by git, but maybe some user will just put the configuration file outside some git repo.

**Comment by LintaoAmons** [comment]
Hi, I'm not very sure if we need add this functionality, cause currently these's a command `ScratchInitConfig`, will call this function

https://github.com/LintaoAmons/scratch.nvim/blob/9b35cf9ff22f243cea72febc37416897d6eae184/lua/scratch/config.lua#L250

And it should be able to reset the configuration.

But yes, it's quit dangerous to rewrite the configuration without warning the user. Previously I was thinking maybe the configuration is tracked by git, but maybe some user will just put the configuration file outside some git repo.

**Comment by Rizhiy** [comment]
I have my configuration tracked by git, so it was already present when I installed the plugin on a new computer.
My configuration was overwritten during initialisation, which is not nice.

Silently overwriting user files is a rather bad behaviour.

File: lua/scratch/config.lua [hunk 1]
```diff
@@ -148,24 +169,20 @@ local function initProcess(force)
...
-  -- write the scratch_file_dir into CONFIG_FILE_PATH file
-  local dir_path = vim.fn.fnamemodify(CONFIG_FILE_FLAG_PATH, ":h")
-  if vim.fn.isdirectory(dir_path) == 0 then
-    vim.fn.mkdir(dir_path, "p")
+  -- check if config already exists
+  if vim.fn.filereadable(configFilePath) == 1 then
+    vim.ui.input({
+      prompt = "Found existing config at " .. configFilePath .. ".\n Do you want overwrite it?(Yes/No)",
+    }, function(input)
+      if input == "Yes" then
```
Comment (Rizhiy) [inline lua/scratch/config.lua:177]: I think this is more standard:
```suggestion
      prompt = "Found existing config at " .. configFilePath .. ".\n Do you want overwrite it? (y/N):",
    }, function(input)
      if input == "y" then
```
One keypress is easier)

File: lua/scratch/config.lua [hunk 2]
```diff
@@ -128,6 +128,27 @@ M.checkInit = function()
...
+local function writeConfig(configFilePath)
+  -- write the scratch_file_dir into CONFIG_FILE_PATH file
+  local dir_path = vim.fn.fnamemodify(CONFIG_FILE_FLAG_PATH, ":h")
+  if vim.fn.isdirectory(dir_path) == 0 then
+    vim.fn.mkdir(dir_path, "p")
+  end
+
+  local file = io.open(CONFIG_FILE_FLAG_PATH, "w")
+  file:write(configFilePath)
+  file:close()
```
Comment (Rizhiy) [inline lua/scratch/config.lua:141]: This needs to be in the main function (`initProcess`), otherwise it will keep asking the user every time vim starts.

**Review by Rizhiy — COMMENTED** [review]
CONFIG_FILE_FLAG_PATH is an internal file, you should not expect user to keep track of it, so need to consider the case where config is already present, but flag is not.

Also, I'm not sure why it is even required, surely you can just ask the user where their config is during init.

**Comment by LintaoAmons** [comment]
> Also, I'm not sure why it is even required, surely you can just ask the user where their config is during init.

Cause in the very beginning I don't want to follow the `setup` function convention, so I created a Flag to persist the config file's location. If the flag is not exist, then the process will start to setup the configuration.

**Comment by Rizhiy** [comment]
Ok, then it might be a better idea to just refactor. I will try to submit a new PR next week.

**Comment by LintaoAmons** [comment]
Thx, the reason I didn't want to use setup function is I want to hot reload the config. So don't need to restart the nvim after add a new filetype to the plugin

**Pull request closed without merging**